Python/Product/Analysis/Infrastructure/UriExtensions.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Net;
+
+namespace Microsoft.PythonTools.Analysis.Infrastructure {
+    public static class UriExtensions {
+        public static string ToAbsolutePath(this Uri uri) {
+            if(IsWindows()) {
+                // VS Code always sends /-based paths. Also, its file URL looks like
+                // file:///C:/foo which yields AbsolutePath as /C:/foo which is wrong.
+                // We need to clean this up.
+                var path = WebUtility.UrlDecode(uri.AbsolutePath).Replace('/', '\\');
```

Comment:
We should use LocalPath to extract local paths, not AbsolutePath. (If there are path forms that do not work well with LocalPath, we can special case those. But the normal case works fine.)

Reply:
OK. LocalPath still needs to be cleaned up. It looks like `/f:/VSCP/src/test/pythonFiles`